### ISSUE TITLE
Wrap Logging::Logger with ActiveSupport::TaggedLogging

### ISSUE THREAD

**Issue by sshaw** [TwP/logging-rails#20]
Any reason not to do this?

**Comment by TwP** [comment]
Logging provides the [nested diagnostic context](https://github.com/TwP/logging/blob/master/lib/logging/diagnostic_context.rb#L230-L266) that does exactly this. You can also used the [mapped diagnostic context](https://github.com/TwP/logging/blob/master/examples/mdc.rb) to attach named attributes to each log message.

```ruby
Logging.ndc.push("fact1") {
  logger.info("some message")
}
```

You can also nest these contexts. The block format is optional (although convenient).

```ruby
Logging.ndc.push("fact1")
# lots of other code
Logging.ndc.push("fact2") {
  logger.info("another message")
}
# and even more code
Logging.ndc.pop
```

**Comment by sshaw** [comment]
> Logging provides the nested diagnostic context that does exactly this. You can also used the mapped diagnostic context

Yes, I'm familiar with these, but why force one  to cook up something custom
just to get what Rails provides out of the box by assigning `Proc`s or `Symbols` to `config.log_tags`?

Of course it's trivial for one to wrap `Logging::Logger` with `ActiveSupport::TaggedLogging`, but 
this is *logging-rails*. It should integrate logging with Rails' features, not the other way around :smile_cat:

**Comment by sshaw** [comment]
Though there is one thing to note: Rails will wrap all loggers (ActiveRecord, ActionController, etc...)
